packages/solana/src/client.ts

```diff
@@ -365,7 +376,8 @@ export class Web3Modal extends Web3ModalScaffold {
           imageUrl: chainImages?.[chain.chainId],
           chain: this.chain
         })
-        if (isConnected && address) {
+
+        if (address) {
```

Comment:
same, why remove the connected check?

Reply:
It has redundancy with `address` and blocks `connect` function for setting `isConnected` only after everything is ok in the connection. Do you think this is not the correct behavior?